src/libraries/Microsoft.PowerFx.Core/Binding/BinderUtils.cs

```diff
@@ -847,5 +848,95 @@ public static BinderCheckTypeResult CheckBinaryOpCore(IErrorContainer errorConta
                     return new BinderCheckTypeResult() { Node = node, NodeType = DType.Error };
             }
         }
+
+        public static bool TryGetConstantValue(CheckTypesContext context, TexlNode node, out string nodeValue)
+        {
+            Contracts.AssertValue(node);
+            nodeValue = null;
+            switch (node.Kind)
+            {
+                case NodeKind.StrLit:
+                    nodeValue = node.AsStrLit().Value;
+                    return true;
+                case NodeKind.BinaryOp:
+                    var binaryOpNode = node.AsBinaryOp();
+                    if (binaryOpNode.Op == BinaryOp.Concat)
+                    {
+                        if (TryGetConstantValue(context, binaryOpNode.Left, out var left) && TryGetConstantValue(context, binaryOpNode.Right, out var right))
+                        {
+                            nodeValue = string.Concat(left, right);
+                            return true;
+                        }
+                    }
+
+                    break;
+                case NodeKind.Call:
+                    var callNode = node.AsCall();
+                    if (callNode.Head.Name.Value == BuiltinFunctionsCore.Concatenate.Name)
+                    {
+                        var parameters = new List<string>();
+                        foreach (var argNode in callNode.Args.Children)
+                        {
+                            if (TryGetConstantValue(context, argNode, out var argValue))
+                            {
+                                parameters.Add(argValue);
+                            }
+                            else
+                            {
+                                break;
+                            }
+                        }
+
+                        if (parameters.Count == callNode.Args.Count)
+                        {
+                            nodeValue = string.Join(string.Empty, parameters);
+                            return true;
+                        }
+                    }
+
+                    break;
+                case NodeKind.FirstName:
+                    // Possibly a non-qualified enum value
+                    var firstNameNode = node.AsFirstName();
+
+                    if (context.NameResolver.Lookup(firstNameNode.Ident.Name, out var firstNameInfo, NameLookupPreferences.None))
```

Comment:
>context.NameResolver.Lookup [](http://example.com/codeflow?start=24&length=27)

This was changed from:
var firstNameInfo = GetInfo(firstNameNode);

I think this is safe and behaves the same.

Reply:
This appears to be safe, as it's passing the Lab in PAClient